internal/packager/fork.go

```diff
@@ -32,16 +32,22 @@ func Fork(originName, forkName, outputDir string, maintainers []string) error {
 
 	// create app dir in output-dir
 	namespace, name := splitPackageName(forkName)
-	appPath := filepath.Join(outputDir, internal.DirNameFromAppName(name))
-	if err := os.MkdirAll(appPath, 0755); err != nil {
+	appDir := filepath.Join(outputDir, internal.DirNameFromAppName(name))
+	if err := os.MkdirAll(appDir, 0755); err != nil {
 		return err
 	}
 
 	// iterate on contents
 	for k, vs := range payload {
 		v := []byte(vs)
-		if strings.Contains(k, "/") || strings.Contains(k, "\\") {
-			log.Infof("dropping payload element with unexpected path separator: %s", k)
+		// Deal with windows/linux slashes
+		convertedFilepath := filepath.FromSlash(k)
+
+		// Check we aren't doing ./../../../ etc in the path
+		fullFilepath := filepath.Join(appDir, convertedFilepath)
+		_, err := filepath.Rel(appDir, fullFilepath)
+		if err != nil {
```

Comment:
This function is basically the same in fork.go and registry.go, there is a minor difference whereby fork alters the metadata file. Should I extract the duplication?

Reply:
Yep I guess you can try yes 😄

Reply:
Ok, i've pushed an attempt at this. There were 2 options: Inject the metadata editing code into the extracted function via a func parameter (ugly), or extract all the files onto disk once then update the metadata file after extraction. I went for the latter as it seemed cleaner, however this is slightly slower. Thoughts?